docs/rules/no-multiple-empty-lines.md

```diff
@@ -94,3 +94,7 @@ var bar = 3;
 ## When Not To Use It
 
 If you do not care about extra blank lines, turn this off.
+
+## Using It With The `eol-last` Rule
+
+If you wish to use this rule with the `eol-last` core rule regarding ending files with a single newline character, `maxEOF` should be set to 0, as an empty line is actually represented by two newline characters (`\n\n`). There is no empty line at the end of a file after the last `\n`, although editors may show an additional line.
```

Comment:
As I understand, the `"maxEOF": 1` will result in the file ending with `\n\n`, which as you mentioned is not what people usually want. I can add this explanation (i.e., saying that if you want precisely one `\n` character at the end, it has to be set to zero) and that it may be redundant if `no-trailing-spaces` and `eol-last` set to `"never"` are used.

Reply:
> As I understand, the `"maxEOF": 1` will result in the file ending with `\n\n`, which as you mentioned is not what people usually want.

It would indeed allow (though not require) files to end with `\n\n`, but that might be what someone really wants: to allow one "real" empty line at the end, for some reason.

Otherwise, we wouldn't need this option as a number, it could have been just a boolean to disallow empty lines at EOF.

I personally think we should just change the existing `maxEOF` section, instead of adding a new one. Maybe we should replace `maxEOF: 1` examples with `maxEOF: 0` examples (which is what probably most people want to set instead of misleading 1), and add some notes about files ending with a newline sequence and editors.

Reply:
I'm wondering would it make sense to also add something like this at the end of the `maxEOF` section, to show how the incorrect and correct example could look in an editor that renders an extra line.

incorrect:

```
1    /*eslint no-multiple-empty-lines: ["error", { "max": 2, "maxEOF": 0 }]*/⏎
2    ⏎
3    var foo = 5;⏎
4    ⏎
5    ⏎
6    var bar = 3;⏎
7    ⏎
8
```

correct:

```
1    /*eslint no-multiple-empty-lines: ["error", { "max": 2, "maxEOF": 0 }]*/⏎
2    ⏎
3    var foo = 5;⏎
4    ⏎
5    ⏎
6    var bar = 3;⏎
7
```

Reply:
Ok, I've made a few changes and added this explanation. See if you think it's clearer or if it could still be improved.